src/etc/inc/auth.inc

```diff
@@ -337,6 +338,14 @@ function local_user_set(&$user, $force_password = false, $userattrs = null)
         $user_shell = $is_admin && !empty($user['shell']) ? $user['shell'] : '/usr/sbin/nologin';
         $user_group = $is_admin ? 'wheel' : 'nobody';
         $user_home = "/home/{$user_name}";
+
+        if (!file_exists($user_shell)) {
```

Comment:
introduces side affects, we're most certainly not going to do this, also nog guaranteed to work when versions don't match for example (in which case more glue might be needed).

https://github.com/opnsense/core/blob/e0e7e0b638afae2b1e71d0e53ea8c41cb9bc3348/src/opnsense/scripts/firmware/install.sh#L41

Reply:
Fair enough, it was just an idea, but I did wonder if it was too risky.